### ISSUE TITLE
Repository structure validation: add checks for funky repo layouts

### ISSUE THREAD

**Issue by davelab6** [fonttools/fontspector#465]
**The original title of his issue was:** `Repo profile: Check for funky repo layouts`

## Observed behaviour

https://github.com/originaltype/kufam/tree/8e613cd512b89bb205fc6b5f20353039f2dacb89 is a funky repo layout

## Expected behaviour

We should have upstream repo checks for things that are unusual about it.

**Comment by felipesanches** [comment]
All of Font Bakery's `"new check proposal"` issues are being either closed or migrated to the [Fontspector issue tracker](https://github.com/fonttools/fontspector/issues), as there's no intention to implement new checks on the Font Bakery codebase anymore.

If you believe this proposal is still valid, please post a message here asking for it to be migrated. Otherwise, on **July 20, 2025** (30 days from now) it will be closed due to inactivity.

**Comment by felipesanches** [comment]
The fontspector issue tracker does not currently use priority/importance labels, but when it was hosted at the fontbakery issue tracker:

* @felipesanches applied "P3 Soon" on Apr 4, 2019

**Comment by simoncozens** [comment]
More generally, I think one day we should have a "repo" profile which checks the font build environment, whereas the "googlefonts" profile checks the binaries/metadata expecting a google/fonts repo. But not yet.

**Comment by felipesanches** [comment]
But I'd suggest treating it as a subset of checks in the googlefonts profile instead of making it a separate profile.

**Comment by simoncozens** [comment]
No, I don't think so. We literally want to run different checks for the downstream and upstream environments, and profiles are how you organise which checks to run.

**Comment by felipesanches** [comment]
Sounds like what one would call a "googlefonts_upstream" profile then... which is pretty much the same as a check in the "googlefonts" profile with a condition `Is this one of our upstream repos?` that only actually runs the check when it is computed as true.

**Comment by felipesanches** [comment]
The semantic issue here is that if you split it out to a new profile, then it blurs a bit the `universal` versus `vendor-specific` spectrum. It is definitely still google-specific, so that's why I think it should still live in there.

**Comment by simoncozens** [comment]
It is google-specific, yes. So we have `googlefonts` and `googlefonts/upstream`.
